### PR TITLE
Add validation for immutable fields.

### LINKED ISSUE #22: Immutable fields are, in fact, not immutable.
I just noticed that I can change the `ipv4`, `ipv6`, and `nodeSelector` fields of the ClusterCIDR resource after provisioning. This seems unintended, as according to the CRD definition, these fields should be immutable.

### PR DESCRIPTION
**Pull request by mneverov**
Closes #22

### CODE REVIEW

File: pkg/controller/ipam/ipam_test.go [hunk 1]
```diff
@@ -168,6 +168,25 @@ var _ = ginkgo.Describe("Pod CIDRs", ginkgo.Ordered, func() {
 		}, gomega.Equal(expectedPodCIDRs3)))
 	})
 
+	ginkgo.It("should fail to update immutable fields", func() {
```
Comment (aojea) [inline pkg/controller/ipam/ipam_test.go:171]: where are these tests running?

should they be running here https://github.com/kubernetes-sigs/node-ipam-controller/actions/runs/10150448361/job/28067570270?pr=24?

it says

> ?   	sigs.k8s.io/node-ipam-controller/pkg/controller/ipam/test	[no test files]

Comment (mneverov) [reply pkg/controller/ipam/ipam_test.go:171]: Tests output should be improved, agree. Ipam package run [output](https://github.com/kubernetes-sigs/node-ipam-controller/actions/runs/10150448361/job/28067570270?pr=24#step:4:145). [Output from](https://github.com/kubernetes-sigs/node-ipam-controller/actions/runs/10150448361/job/28067570270?pr=24#step:4:237) the controller for ClusterCIDR with the key `validate-immutable`

**Comment by aojea** [comment]
/lgtm
/approve

Thanks

File: pkg/controller/ipam/ipam_test.go (hunk 1 above)
Comment (aojea) [reply pkg/controller/ipam/ipam_test.go:171]: not familiar with this test suite, thanks for the clarification

**Pull request merged**